app/controllers/admin_controller.rb

```diff
@@ -0,0 +1,13 @@
+class AdminController < ApplicationController
+before_action :authorize_admin!
+
+  def index; end
+
+  private
+    def authorize_admin!
+      authenticate_user!
+      unless current_user.admin?
+        redirect_to root_path, alert: 'You must be an admin to do that.'
+     end
+  end
```

Comment:
I don't think you need to call authenticate_user! here let's remove extra spaces, add a line after `private` like:
```suggestion
  
  def authorize_admin!
    return if current_user.admin?
    
    redirect_to root_path, alert: 'You must be an admin to do that.'
  end
```